src/main.ts

```diff
@@ -64,6 +65,61 @@ async function main() {
 		console.log(`Logged in as ${client.user?.tag}`);
 	});
 
+	client.on('messageReactionAdd', async (reaction, user) => {
+		// When a reaction is received, check if the structure is partial
+		if (reaction.partial) {
+			// If the message this reaction belongs to was removed, the fetching might result in an API error which should be handled
+			try {
+				await reaction.fetch();
+			} catch (error) {
+				log.error(client, {message: 'Something went wrong when fetching the message:',stack:'',name:'React Message Error'});
+				// Return as `reaction.message.author` may be undefined/null
+				return;
+			}
+		}
+
+
+	
+		for(const res of config.reaction_role_response) {
+			if(res.message_id !== reaction.message.id) continue;
+			for(const roles of res.roles) {
+				if(roles.emoji_name !== reaction.emoji.name) continue;
+				const guildreactuser = reaction.message.guild?.members.cache.get(user.id);
+				guildreactuser?.roles.add(roles.role_id);
+				break;
+			}
+			break;
```

Comment:
This break makes the outer for pointless, if you need to break out of both fors from inside the loop use a label